mlir/lib/Conversion/VectorToSPIRV/VectorToSPIRV.cpp

```diff
@@ -220,6 +220,34 @@ struct VectorFmaOpConvert final : public OpConversionPattern<vector::FMAOp> {
   }
 };
 
+struct VectorFromElementsOpConvert final
+    : public OpConversionPattern<vector::FromElementsOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(vector::FromElementsOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Type resultType = getTypeConverter()->convertType(op.getType());
+    auto elements = op.getElements();
+    if (!resultType)
+      return failure();
+    if (isa<spirv::ScalarType>(resultType)) {
+      // In the case with a single scalar operand / single-element result,
+      // pass through the scalar.
+      rewriter.replaceOp(op, elements[0]);
+      return success();
+    } else if (cast<VectorType>(resultType).getRank() == 1) {
```

Comment:
We don't put `else` after `return`: https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return

Also, I think we don't need this if because there are only two cases: scalar and 1d with 2 or more elements. Everything else should be rejected by the type converter.

Reply:
See here, I was trying to pre-empt this comment: https://github.com/llvm/llvm-project/pull/118540#discussion_r1868331381

I guess I was afraid of using an `assert`, but maybe it's fine. Does this look better now?